File: fizzbuzz/fizzbuzz.cpp
// Build:
// $ clang++ -std=c++17 fizzbuzz.cpp
// Run:
// $ ./a.out

#include <vector>
#include <iostream>
#include <string>

auto fizzbuzz() -> std::vector<std::string> {
    std::vector<std::string> result;
    for (int i = 1; i < (100 + 1); i++){
        if(i % 5 == 0 && i % 3 == 0){
            result.push_back("FizzBuzz");
        }
        else if(i % 5 == 0){
            result.push_back("Buzz");
        }
        else if(i % 3 == 0){
            result.push_back("Fizz");
        }
        else{
            result.push_back(std::to_string(i));
        }
    }
    return result;
}

int main()
{
    auto vec = fizzbuzz();
    for (const auto& s : vec) {
        std::cout << s << std::endl;
    }
}

// 参考サイト:
// https://www.javaer101.com/article/7698965.html

File: fizzbuzz/fizzbuzz.rs
// Build:
// $ rustc fizzbuzz.rs
// Run:
// $ ./fizzbuzz

const N: usize = 100;

fn fizzbuzz() -> Vec<String> {
    let mut result: Vec<String> = Vec::with_capacity(N);
    for i in 1..=N {
        result.push(match (i % 3, i % 5) {
            (0, 0) => "FizzBuzz".into(),
            (0, _) => "Fizz".into(),
            (_, 0) => "Buzz".into(),
            (_, _) => i.to_string(),
        })
    }
    result
}

fn main() {
    println!("{:?}", fizzbuzz());
}

// 参考サイト:
// https://gist.github.com/mre/f9e8d31cd1bde76fc714cae3f9d4e131
